prog/weaver/main.go

```diff
@@ -246,6 +253,19 @@ func main() {
 	}
 	config.ProtocolMinVersion = byte(protocolMinVersion)
 
+	if !proxyConfig.Disabled {
+		// Start Weave Proxy:
+		proxy, err := weaveproxy.NewProxy(*proxyConfig)
+		if err != nil {
+			Log.Fatalf("Could not start proxy: %s", err)
+		}
+		defer proxy.Stop()
+		listeners := proxy.Listen()
```

Comment:
Nitpick, but maybe it makes sense to create `proxy.Start()` and to  move the rest of the lines to this method?